### PR TITLE
Account page plus minor issue fixes

### PR DESCRIPTION
**Pull request by alanlgirard**
New account page layout/content to close user story #294. This is the last item to officially declare MVP.  
- Shows the logged-in user's created ideas, backed ideas, and ideas that they are a part of the team.  
- Added functions getUserIdeasById and getUserBacksById to ideaSvc. 
- Changed it so that if a user logs out while on the account page it routes them to the home page.

account-page

Closes favicon transparency Issue #319 
image

Closes top idea likes issue #315 
likes 20

@YashdalfTheGray @Mctalian @monotkate @bcbrennecke 

- [ ] Code Review {@YashdalfTheGray}
- [ ] Code Review {everyone else}
- [x] Functional Review {@YashdalfTheGray}
- [x] Functional Review {@davethenipper}

### CODE REVIEW

File: src/app.css [hunk 1]
```diff
@@ -93,7 +93,7 @@ h1.no-margins {
     -webkit-margin-before: 0;
 }
 
-h2.no-bottom-margin {
+h1.no-bottom-margin, h2.no-bottom-margin, h3.no-bottom-margin, h4.no-bottom-margin {
```
Comment (YashdalfTheGray) [inline src/app.css]: This is more of an understanding question but what are these classes for?

File: src/ideas/ideaItem/ideaItem.tpl.html [hunk 2]
```diff
@@ -10,7 +10,7 @@
             <div class="accent-text author-accent">{{idea.author.name}}</div>
         </div>
     </div>
-    <div flex layout="row" layout-align="center center" class="extra-padding-left-md">
+    <div flex="20" layout="row" layout-align="center center">
```
Comment (YashdalfTheGray) [inline src/ideas/ideaItem/ideaItem.tpl.html]: Think this might be enough for even if something gets triple digit likes.

File: src/app.css (hunk 1 above)
Comment (alanlgirard) [inline src/app.css]: They were when I was playing around with layout stuff.  I can remove them but they could be helpful for the future.

File: src/ideas/ideaItem/ideaItem.tpl.html (hunk 2 above)
Comment (alanlgirard) [inline src/ideas/ideaItem/ideaItem.tpl.html]: Yes I checked triple digits and it works fine.  I just didn't get a screenshot

File: src/app.css (hunk 1 above)
Comment (YashdalfTheGray) [inline src/app.css]: Guess if they're not hurting anything, we can keep them. So they just remove bottom margins from `hx` elements?

File: src/ideas/ideaItem/ideaItem.tpl.html (hunk 2 above)
Comment (YashdalfTheGray) [inline src/ideas/ideaItem/ideaItem.tpl.html]: :+1:

File: src/app.css (hunk 1 above)
Comment (alanlgirard) [inline src/app.css]: Yes they just remove the bottom margin. We were already using it for a h2 on the idea page I think so I just expanded on those while I was messing around so I didn't have to go back and forth adding/removing them

File: src/ideas/ideaItem/ideaItem.tpl.html (hunk 2 above)
Comment (alanlgirard) [inline src/ideas/ideaItem/ideaItem.tpl.html]: Now 4 digits.....Hopefully we get to that issue in the future!

File: src/navigation/toolbar.js [hunk 3]
```diff
@@ -86,6 +86,9 @@ angular.module('flintAndSteel')
                 else if ($state.includes('addidea')) {
```
Comment (YashdalfTheGray) [inline src/navigation/toolbar.js]: Can we change this to
```javascript
else if ($state.includes('addidea') || $state.includes('account') {
    $state.go('home');
}
```

File: src/app.css (hunk 1 above)
Comment (YashdalfTheGray) [inline src/app.css]: Okay, that's fine!

File: src/ideas/ideaItem/ideaItem.tpl.html (hunk 2 above)
Comment (YashdalfTheGray) [inline src/ideas/ideaItem/ideaItem.tpl.html]: fingers crossed!

File: src/users/accountView/accountView.tpl.html [hunk 4]
```diff
@@ -1,12 +1,208 @@
-<md-content layout="column" layout-align="start center">
-    <h1>Account</h1>
-    <div>
-        <!-- Need to figure out page layout -->
-        <div layout="row" layout-align="center">
-            <identicon username='user.username' size='130'></identicon>
+<md-content class="md-padding">
+        <div class="md-padding" layout="row" layout-align="start center" flex>
```
Comment (YashdalfTheGray) [inline src/users/accountView/accountView.tpl.html]: I think you might have this indented too far.

File: src/ideas/ideaItem/ideaItem.tpl.html (hunk 2 above)
Comment (monotkate) [inline src/ideas/ideaItem/ideaItem.tpl.html]: ha!  Love it.  That was my first thought.  3, not 4 lol

**Comment by YashdalfTheGray** [comment]
I know that this might be a little out of the scope of this but how's the account page look on mobile?

File: src/ideas/ideaSvc/ideaSvc.spec.js [hunk 5]
```diff
@@ -338,6 +350,36 @@ describe('ideaSvc', function() {
...
+
+            it('should return false if an id was not supplied', function() {
+                ideaSvc.getUserIdeasById().then(function(result) {
+                    expect(result).toBe(false);
+                });
+
+            });
+        });
+
+        describe('ideaSvc.getUserBacksById', function() {
```
Comment (monotkate) [inline src/ideas/ideaSvc/ideaSvc.spec.js]: Can we test an idea that hasn't backed as well?  Same for ideas?

**Comment by alanlgirard** [comment]
On mobile I remove the identicon and the ideas,backs,teams numbers as seen bellow. I am not a huge fan of the layout of the text for mobile but that can be addressed in our UX talk

image

**Comment by YashdalfTheGray** [comment]
When I run the tests (maybe you don't see this) but there is a warning that pops up that says
```
WARN: 'Deprecation warning: moment construction falls back to js Date. This is discouraged and will be removed in upcoming major release. Please refer to https://github.com/moment/moment/issues/1407 for more info.
undefined'
```
Can you look into that too?

**Comment by YashdalfTheGray** [comment]
Holy shit, well done sir. That's a lot better that most of the other areas of the website! 

Sidenote, do you think you could throw in the fix for #320 while you're at it? I know it's a quick fix which is why I'm telling you, otherwise we can get to it on Sunday. I totally understand #ScopeCreep :scream:

**Comment by monotkate** [comment]
@YashdalfTheGray is officially no longer allowed to yell at me for scope creep in a PR

**Comment by YashdalfTheGray** [comment]
Shush, I asked because I know it's a quick fix but you're right, we should probably handle it on Sunday.

**Comment by YashdalfTheGray** [comment]
Pulled it down, looks good. Pending some code comments.

File: src/ideas/ideaSvc/ideaSvc.js [hunk 6]
```diff
@@ -158,6 +158,24 @@ angular.module('flintAndSteel')
                 });
             };
 
+            this.getUserIdeasById = function getUserIdeasById(userId) {
+                if (userId) {
+                    return $http.get('/api/v1/ideas/search?forterm=' + userId + '&inpath=authorId');
+                }
+                else {
+                    return $q.when(false);
```
Comment (YashdalfTheGray) [inline src/ideas/ideaSvc/ideaSvc.js]: If you want, replace this with `return $q.reject({ error: 'no userId provided' });`

File: src/ideas/ideaSvc/ideaSvc.js [hunk 7]
```diff
@@ -158,6 +158,24 @@ angular.module('flintAndSteel')
...
+                }
+            };
+
+            this.getUserBacksById = function getUserBacksById(userId) {
+                if (userId) {
+                    return $http.get('/api/v1/ideas/search?forterm=' + userId + '&inpath=backs.authorId');
+                }
+                else {
+                    return $q.when(false);
```
Comment (YashdalfTheGray) [inline src/ideas/ideaSvc/ideaSvc.js]: Replace with `$q.reject()` here too.

File: src/users/accountView/accountView.js [hunk 8]
```diff
@@ -12,40 +14,53 @@ angular.module('flintAndSteel')
...
+                        });
+                    }
+                );
 
-                toastSvc.show(accountName + ' has been logged out!', { duration: 5000 });
-
-                $state.go('home');
-            };
+                //Get user Backs and Teams
+                ideaSvc.getUserBacksById(userId).then(
```
Comment (YashdalfTheGray) [inline src/users/accountView/accountView.js]: There should probably be `.catch()` here for promise rejection scenarios.

File: src/users/accountView/accountView.js [hunk 9]
```diff
@@ -12,40 +14,53 @@ angular.module('flintAndSteel')
...
+                                function(result) {
+                                    result.data.timeCreated = moment(new Date(result.data.timeCreated)).calendar();
+                                    result.data.timeModified = moment(new Date(result.data.timeModified)).calendar();
+                                    $scope.userBacks.push(result.data);
+                                    _.some(result.data.team, function(member) {
+                                        if (member.memberId === userId) {
+                                            $scope.userTeams.push(result.data);
+                                        }
+                                    });
+                                });
```
Comment (YashdalfTheGray) [inline src/users/accountView/accountView.js]: Misaligned brackets.

**Comment by davethenipper** [comment]
Functional Review Done.  I believe the Team Code Reviewed this?

**Comment by YashdalfTheGray** [comment]
Yeah, I believe we're waiting on @alanlgirard to get to the last pieces of this.

**Comment by YashdalfTheGray** [comment]
This all good to be merged?

**Comment by alanlgirard** [comment]
Yes this should be good to be merged.  I was just waiting for @YashdalfTheGray or @monotkate approval to merge.

**Comment by YashdalfTheGray** [comment]
You can merge this. I think I'll tag MVP tonight.

**Pull request merged**